docs/en/query_language/agg_functions/reference.md

```diff
@@ -906,32 +906,510 @@ SELECT quantileTiming(0.5)(number / 2) FROM numbers(10)
 
 Differs from the `quantileTiming` function in that it has a second argument, "weights". Weight is a non-negative integer.
 The result is calculated as if the `x` value were passed `weight` number of times to the `quantileTiming` function.
 
-## quantileExact(level)(x)
+## quantileExact(level)(x) {#quantileexact}
 
 Computes the quantile of 'level' exactly. To do this, all the passed values ​​are combined into an array, which is then partially sorted. Therefore, the function consumes O(n) memory, where 'n' is the number of values that were passed. However, for a small number of values, the function is very effective.
 
-## quantileExactWeighted(level)(x, weight)
+## quantileExactWeighted(level)(x, weight) {#quantileexactweighted}
 
 Computes the quantile of 'level' exactly. In addition, each value is counted with its weight, as if it is present 'weight' times. The arguments of the function can be considered as histograms, where the value 'x' corresponds to a histogram "column" of the height 'weight', and the function itself can be considered as a summation of histograms.
 
 A hash table is used as the algorithm. Because of this, if the passed values ​​are frequently repeated, the function consumes less RAM than `quantileExact`. You can use this function instead of `quantileExact` and specify the weight as 1.
 
-## quantileTDigest(level)(x)
+## quantileTDigest(level)(x) {#quantiletdigest}
 
 Approximates the quantile level using the [t-digest](https://github.com/tdunning/t-digest/blob/master/docs/t-digest-paper/histo.pdf) algorithm. The maximum error is 1%. Memory consumption by State is proportional to the logarithm of the number of passed values.
 
 The performance of the function is lower than for `quantile` or `quantileTiming`. In terms of the ratio of State size to precision, this function is much better than `quantile`.
 
 The result depends on the order of running the query, and is nondeterministic.
 
-## median(x)
+## median {#median}
 
-All the quantile functions have corresponding median functions: `median`, `medianDeterministic`, `medianTiming`, `medianTimingWeighted`, `medianExact`, `medianExactWeighted`, `medianTDigest`. They are synonyms and their behavior is identical.
+Computes median or 2-quantile of numeric data set. Median is a middle number of sequence, that separates higher half from the lower half of this sequence. If number of values in the data set is odd then mediana usually is an arithmetic mean of two middle values.
+
+This function based on [reservoir sampling](https://en.wikipedia.org/wiki/Reservoir_sampling) method with a reservoir size up to 8192. Since this algorithm uses a random number generator this function is non-deterministic.
+
+This algorithm provides very low accuracy
+
+**Syntax** 
+
+```sql
+median(x);
+```
+
+Alias: `quantile`. 
+
+Median is an alias of `quantile` function with an argument of 0.5 (50%).
+
+**Parameters** 
+
+- `x` — numeric data types, Date and DateTime.
```

Comment:
number. [Expression](link-to-expressions) resulting in numeric [data types](link to introduction to data types), [Date](link to date) and [DateTime](link to datetime).